mypy/newsemanal/semanal.py

```diff
@@ -491,8 +491,14 @@ def file_context(self, file_node: MypyFile, fnam: str, options: Options,
         del self.options
 
     def visit_func_def(self, defn: FuncDef) -> None:
-        if not defn.is_decorated and not defn.is_overload:
-            self.add_func_to_symbol_table(defn)
+        defn.is_conditional = self.block_depth[-1] > 0
+
+        # Add module-level function to symbol table on first iteration
+        # (recurse_into_functions == False). Also add nested functions
+        # always, since they won't be processed as separate targets.
+        if not self.recurse_into_functions or len(self.function_stack) > 0:
```

Comment:
This explanation is not very clear. I would rather say that we don't add function top-level functions when we just enter each of it during second phase of analysis -- processing of functions.

Also I would rather explain _why_ this is needed (I agree that this looks reasonable).